tools/link.py

```diff
@@ -1986,6 +1986,8 @@ def run_embind_gen(wasm_target, js_syms, extra_settings):
 @ToolchainProfiler.profile_block('emit tsd')
 def phase_emit_tsd(options, wasm_target, js_syms, metadata):
   logger.debug('emit tsd')
+  if options.oformat in (OFormat.WASM, OFormat.BARE):
+    exit_with_error('Wasm only output is not compatible --emit-tsd')
```

Comment:
Could you report this error earlier when you set `settings.EMIT_TSD = True` in phase_linker_setup.

I like to report this linker errors as early as possible.